include/execpools/tbb/tbb_thread_pool.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 Ben FrantzDale
+ * Copyright (c) 2021-2023 Facebook, Inc. and its affiliates.
+ * Copyright (c) 2021-2024 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <tbb/task_arena.h>
+
+#include <exec/static_thread_pool.hpp>
+#include <execpools/thread_pool_base.hpp>
+
+namespace execpools {
+
+  class tbb_thread_pool : public thread_pool_base<tbb_thread_pool> {
+   public:
+    //! Constructor forwards to tbb::task_arena constructor:
+    template <class... Args>
+      requires stdexec::constructible_from<tbb::task_arena, Args...>
+    explicit tbb_thread_pool(Args&&... args)
+      : arena_{std::forward<Args>(args)...} {
+      arena_.initialize();
+    }
+
+    [[nodiscard]]
+    auto available_parallelism() const -> std::uint32_t {
```

Comment:
This isn't part of P2300 is it? It's just a nice-to-have, I think?

Reply:
I couldn't find any reference about it. But from my personal point of view it is nice to have but also I can imagine that not 'all' of the threading libraries has this information or easy to access. I.e.: In asio_thread_pool the `_executor` member is required due to fulfill this functional requirement.)
Thus, if we removing it then the thread pool will be more generic.